AppCenterIdentity/AppCenterIdentity/MSIdentity.m

```diff
@@ -131,26 +144,132 @@ + (void)handleUrlResponse:(NSURL *)url {
-                                  // TODO: Implement error handling.
-                                  if (e) {
-                                  } else {
-                                    NSString __unused *accountIdentifier = result.account.homeAccountId.identifier;
-                                    self.accessToken = result.accessToken;
-                                  }
-                                }];
 }
+
+#pragma mark - Private methods
+
+- (NSString *)identityConfigFilePath {
+  return [NSString stringWithFormat:@"%@/%@", kMSIdentityPathComponent, kMSIdentityConfigFilename];
+}
+
+- (BOOL)loadConfigurationFromCache {
+  NSData *configData = [MSUtility loadDataForPathComponent:[self identityConfigFilePath]];
+  self.identityConfig = [self deserializeData:configData];
+  if (self.identityConfig == nil || ![self.identityConfig isValid]) {
+    [self clearConfigurationCache];
+    self.identityConfig = nil;
+    return NO;
+  }
+  return YES;
+}
+
+- (void)downloadConfigurationWithETag:(nullable NSString *)eTag {
+
+  // Download configuration.
+  MSIdentityConfigIngestion *ingestion =
+      [[MSIdentityConfigIngestion alloc] initWithBaseUrl:@"https://mobilecentersdkdev.blob.core.windows.net" appSecret:self.appSecret];
+  [ingestion sendAsync:nil
+                   eTag:eTag
+      completionHandler:^(__unused NSString *callId, NSHTTPURLResponse *response, NSData *data, __unused NSError *error) {
+        MSIdentityConfig *config = nil;
+        if (response.statusCode == MSHTTPCodesNo304NotModified) {
+          MSLogInfo([MSIdentity logTag], @"Identity configuration hasn't changed.");
+        } else if (response.statusCode == MSHTTPCodesNo200OK) {
+          config = [self deserializeData:data];
+          if ([config isValid]) {
+            NSURL *configUrl = [MSUtility createFileAtPathComponent:[self identityConfigFilePath]
+                                                           withData:data
+                                                         atomically:YES
+                                                     forceOverwrite:YES];
+
+            // Store eTag only when the configuration file is created successfully.
+            if (configUrl) {
+
+              // Response header keys are case-insensitive but NSHTTPURLResponse contains case sensitive keys in Dictionary.
+              for (NSString *key in response.allHeaderFields.allKeys) {
+                if ([[key lowercaseString] isEqualToString:kMSETagResponseHeader]) {
+                  [MS_USER_DEFAULTS setObject:(_Nonnull id)response.allHeaderFields[key] forKey:kMSIdentityETagKey];
+                  break;
+                }
+              }
+            } else {
+              MSLogWarning([MSIdentity logTag], @"Couldn't create Identity config file.");
+            }
+            self.identityConfig = config;
+
+            // Reinitialize client application.
+            [self configAuthenticationClient];
```

Comment:
Why is the app re-configured here? I imagine if the user has already signed in, they won't need the app to be reconfigured, and I am worried about the side effects of doing this at that time.

Reply:
If the configuration has changed, it should use the latest configuration to get user information. Not sure there will be side effects on this but at least this aligns to Android implementation. We can revisit this one later.

Reply:
You need to use latest config in case the API will be called after, also keep in mind we will have more than just login flow.

Reply:
Ah true. Okay, just wanted to understand a bit better